starter-code/src/memory.js

```diff
@@ -1,12 +1,39 @@
 var MemoryGame = function (cards) {
   this.cards = cards;
+  this.pickedCards = [];
+  this.pairsClicked = 0;
+  this.pairsGuessed = 0;
 };
 
-MemoryGame.prototype.shuffleCards = function () {
-};
+MemoryGame.prototype.shuffleCards = function(cards) {
+  var newArrayOfCards = [];
+  var fixedLength = cards.length; 
+  for (i=0; i<fixedLength; i++){
+    var chooseRandomCard = Math.floor(Math.random()*(cards.length));
+    
+    newArrayOfCards.push(cards[chooseRandomCard]);
```

Comment:
este for solo re esta regresando 7 cartas y las demás las coloca como "undefined" 

yo te recomendara hacer esto ` const j = Math.floor(Math.random() * (i + 1))`esto te regresara un elemento en una al azar y combinado con esto `[cards[i], cards[j]] = [cards[j], cards[i]];` sustituiremos en la posición i colocando al nuevo elemento j,  de esta forma podremos tener el arreglo ya con los tacos mixeados 

no es necesario hacer un return  ni el splice